pallets/nfts/src/features/atomic_swap.rs

```diff
@@ -209,10 +209,15 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		}
 
 		// This also removes the swap.
-		Self::do_transfer(send_collection_id, send_item_id, receive_item.owner.clone(), |_, _| {
-			Ok(())
-		})?;
 		Self::do_transfer(
+			caller,
+			send_collection_id,
+			send_item_id,
+			receive_item.owner.clone(),
+			|_, _| Ok(()),
+		)?;
+		Self::do_transfer(
+			receive_item.owner.clone(),
```

Comment:
Shouldnt this be caller as well? The caller would be the account concluding the swap I assume, so they should pay for the deposit if the other party potentially does not yet have an account balance of the swapped collection and does not have sufficient funds to pay for the account balance deposit.